docs2/site/docs/migrations/migration8.md

```diff
@@ -288,6 +288,33 @@ private class Query
 }
 ```
 
+### 8. Validation rules can read or validate field arguments and directive arguments
+
+Validation rules can now execute validation code either before or after field arguments
+have been read. This is useful for edge cases, such as when a complexity analyzer needs
+to read the value of a field argument to determine the complexity of the field.
+
+The `ValidateAsync` method on `IValidationRule` has been changed to `GetPreNodeVisitorAsync`,
+and a new method `GetPostNodeVisitorAsync` has been added. Also, the `IVariableVisitorProvider`
+interface has been combined with `IValidationRule` and now has a new method `GetVariableVisitorAsync`.
+So the new `IValidationRule` interface looks like this:
+
+```csharp
+public interface IValidationRule
+{
+    ValueTask<INodeVisitor?> GetPreNodeVisitorAsync(ValidationContext context);
+    ValueTask<IVariableVisitor?> GetVariableVisitorAsync(ValidationContext context);
+    ValueTask<INodeVisitor?> GetPostNodeVisitorAsync(ValidationContext context);
+}
+```
+
+This allows for a single validation rule to validate AST structure, validate variable values,
+and/or validate coerced field and directive arguments.
+
+To simplify the creation of validation rules, the `ValidationRuleBase` class has been added,
+which implements the `IValidationRule` interface and provides default implementations for
+all three methods.
+
```

Comment:
Are you going to include the comparison with `Validator` and `Parser` delegates mentioned in the previous PR? I think it was important. https://github.com/graphql-dotnet/graphql-dotnet/pull/3873

Reply:
I've added the new documentation for creating custom validation rules into the official GraphQL.NET docs (none existed before), referenced from the migration document.  Ok?  It includes various samples you mentioned.